openskill/rate.py

```diff
@@ -216,22 +217,10 @@ def predict_win(teams: List[List[Rating]], **options) -> List[Union[int, float]]
             )
         )
 
-    if n > 2:
-        cache = deque(pairwise_probabilities)
-        probabilities = []
-        partial = len(pairwise_probabilities) / n
-        while len(cache) > 0:
-            aggregate = []
-            for length in range(int(partial)):
-                aggregate.append(cache.popleft())
-            aggregate_sum = sum(aggregate)
-            aggregate_multiple = n
-            for length in range(1, n - 2):
-                aggregate_multiple *= n - length
-            probabilities.append(aggregate_sum / aggregate_multiple)
-        return probabilities
-    else:
-        return pairwise_probabilities
+    return [
+        (sum(team_prob) / denom) for team_prob in itertools.zip_longest(*[iter(pairwise_probabilities)] * (n-1))
```

Comment:
```python
itertools.zip_longest(*[iter(pairwise_probabilities)] * (n-1))
```

this chunks the `pairwise_probabilities` list in n-1 chunks, which comes from https://stackoverflow.com/questions/312443/how-do-you-split-a-list-into-evenly-sized-chunks

Reply:
(i'm not very good at python, but i didn't know how to better do this)

Reply:
Looks good to me!